docs/tls/CONFIGURATION.md

```diff
@@ -0,0 +1,173 @@
+# Configuring NLK to use TLS / mTLS guide
+
+This guide describes the steps required to configure NLK to use self-signed certificates for TLS / mTLS.
+
+In this context TLS refers to one-way TLS, and mTLS refers to mutual TLS. 
+
+One-way TLS is used to encrypt the traffic between NLK and NGINX Plus, and where NLK validates the NGINX Plus server, 
+but the server _does not_ validate NLK.
+
+Mutual TLS is used to encrypt the traffic between NLK and NGINX Plus, and where NLK validates the NGINX Plus server
+and the server validates NLK.
+
+Mutual TLS is preferred as it provides a higher level of security, both parties know they are talking to a trusted peer.
+
+There are two aspects to configuring NLK to use TLS / mTLS:
+
+1. Providing the required certificates to NLK
+2. Providing the required certificates to NGINX Plus and configuring client authentication
+
+## Configuring NLK
+
+NLK is configured via a ConfigMap. The ConfigMap is named `nlk-config` and is located in the `nlk` namespace. 
+The manifest is located in the `/deployments/deployment` directory of the NLK repository.
+
+### ConfigMap fields
+
+Depending on which TLS / mTLS mode is chosen, the following fields will need to be updated in the NLK ConfigMap:
+
+1. mode
+2. caCertificate
+3. clientCertificate
+
+#### mode
+
+If present, determines the TLS / mTLS mode to use. Valid values are `none`, `tls` and `mtls`.
+If not present, defaults to `none`.
+
+#### caCertificate
+
+Required when `mode` is set to `tls` or `mtls`. The name of the Kubernetes Secret containing the CA certificate.
+
+#### clientCertificate
+
+Required when `mode` is set to `mtls`. The name of the Kubernetes Secret containing the client certificate.
+
+#### Sample ConfigMap
+
+The following is an example of a ConfigMap for NLK:
```

Comment:
I assume the sample config is being shown to indicate to the reader this that this is the thing the following steps will help them build

Reply:
Expanded the section to include more detail about using the sample config